### PR TITLE
feat(sidebar): dynamically generate from mdx pages

### PR DESCRIPTION
**Pull request by dagoof**

### CODE REVIEW

File: docs/plugins/mdx-pages/gatsby-node.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+// Relate an MDX node with the page that we create below.
+//
+// Tried to create these nodes inside the createPages callback,
+// but they don't have an effect unless we also create a node field here.
+// for some reason.
+// not sure.
+const onCreateNode = ({ node, actions, getNode }) => {
+  const { createNodeField } = actions
+
+  if (node.internal.type !== 'Mdx') {
```
Comment (jonathaningram) [inline docs/plugins/mdx-pages/gatsby-node.js]: Oh `Mdx`? Not `mdx`?

File: docs/src/components/sidenav.js [hunk 2]
```diff
@@ -29,117 +28,114 @@ const NavItem = styled.li`
...
+const Contents = ({
+  container: Container = NestedNav,
+  pagePath,
+  items = [],
+  depth = 0,
+}) =>
+  items.length && depth < MAXDepth ? (
+    <Container>
+      {items.map((item, i) => (
+        <Fragment key={i}>
```
Comment (jonathaningram) [inline docs/src/components/sidenav.js]: `key={i}` -> `key={item.id}` or `key={item.url}`

File: docs/src/components/sidenav.js [hunk 3]
```diff
@@ -29,117 +28,114 @@ const NavItem = styled.li`
...
-        {data.pages.edges.map(({ page }, i) => (
-          <NavItem key={i}>
-            <Link to={page.path}>{page.path}</Link>
-          </NavItem>
-        ))}
-      </Nav>
-    )}
+    render={data => {
+      const Option = ({ children }) => {
+        const edge = data.mdxPages.edges.find(
```
Comment (jonathaningram) [inline docs/src/components/sidenav.js]: We'll need the `core-js` find polyfill if we want IE support this code runs in the browser.

File: docs/src/content/installation.mdx [hunk 4]
```diff
@@ -1,16 +1,16 @@
...
 
 Install the client libraries in your choice of programming language
 
 <Examples path="installation"/>
 
-## Unsupported languages
+# Unsupported languages
 
 We detail how to roll-your-own client in the [Getting
 started](getting-started.md) section. Together with the `cURL` examples on
```
Comment (jonathaningram) [inline docs/src/content/installation.mdx]: `getting-started.md` is a broken link.

**Comment by jonathaningram** [comment]
This is awesome.

File: docs/plugins/mdx-pages/gatsby-node.js (hunk 1 above)
Comment (dagoof) [reply docs/plugins/mdx-pages/gatsby-node.js]: this is the node type generated by gatsby-mdx. it's referenced here by them as well

https://github.com/ChristopherBiscardi/gatsby-mdx/blob/master/packages/gatsby-mdx/gatsby/extend-node-type.js#L34

File: docs/src/components/sidenav.js (hunk 3 above)
Comment (dagoof) [reply docs/src/components/sidenav.js]: good pickup. might need to do in following prs as this code gets mangled

**Pull request merged**